### PR TITLE
Rn next merge

### PR DESCRIPTION
**Pull request by imjoehaines**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,6 @@
     "@rollup/plugin-typescript": "^11.0.0",
     "@types/jest": "^29.5.0",
     "@types/node": "^18.11.18",
-    "@types/react": "^18.2.20",
```
Comment (imjoehaines) [inline package.json:28]: These have moved to `packages/platforms/react-native` as they aren't needed by other packages

File: package.json [hunk 2]
```diff
@@ -40,9 +39,7 @@
     "jest": "^29.5.0",
     "jest-environment-jsdom": "^29.5.0",
     "lerna": "^6.4.0",
-    "react": "^18.2.0",
-    "react-native": "^0.72.3",
-    "rollup": "^3.18.0",
+    "rollup": "^3.23.0",
```
Comment (imjoehaines) [inline package.json:42]: We need at least `v3.23.0` for the `prebuilt-chunk` type used here:

https://github.com/bugsnag/bugsnag-js-performance/blob/d1a7bff635829b206a9c57e79a6b445175bff61b/packages/platforms/react-native/rollup.config.mjs#L21-L26

File: packages/delivery-fetch/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bugsnag/delivery-fetch-performance",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
Comment (imjoehaines) [inline packages/delivery-fetch/package.json:3]: Bumped these to match the `lerna.json` version number

File: packages/platforms/react-native/lib/auto-instrumentation/app-start-plugin.tsx [hunk 4]
```diff
@@ -40,24 +40,32 @@ export class AppStartPlugin implements Plugin<ReactNativeConfiguration> {
...
         const WrapperComponent = configuration.wrapperComponentProvider(appParams)
+
         return (
           <AppStartWrapper>
+            {/* 'WrapperComponent' has the wrong type here because of
+              * conflicting react type definitions caused by having both
+              * react-native and react-dom (via the react router integration)
+              * installed
+              */}
+            {/* @ts-ignore-error PLAT-xxxx */}
```
Comment (imjoehaines) [inline packages/platforms/react-native/lib/auto-instrumentation/app-start-plugin.tsx:62]: TODO: make a ticket to fix this

File: packages/platforms/react-native/rollup.config.mjs [hunk 5]
```diff
@@ -20,7 +20,6 @@ function reactNativeSpecPlugin () {
     buildStart () { 
       this.emitFile({
         type: 'prebuilt-chunk',
-        id: 'lib/NativeBugsnagPerformance.ts',
```
Comment (imjoehaines) [inline packages/platforms/react-native/rollup.config.mjs:23]: `id` isn't valid for a `prebuilt-chunk`: https://rollupjs.org/plugin-development/#this-emitfile

File: .eslintrc.js [hunk 6]
```diff
@@ -95,6 +95,9 @@ module.exports = {
       rules: {
         ...tsRuleOverrides,
         ...jestRuleOverrides
+      },
+      parserOptions: {
+        project: './tsconfig.json'
```
Comment (imjoehaines) [inline .eslintrc.js:100]: Required for the new versions of typescript-eslint packages (I think this should always have been there but didn't cause errors before)

File: package.json [hunk 7]
```diff
@@ -25,26 +25,23 @@
     "@rollup/plugin-typescript": "^11.0.0",
     "@types/jest": "^29.5.0",
     "@types/node": "^18.11.18",
-    "@types/react": "^18.2.20",
-    "@typescript-eslint/eslint-plugin": "^5.50.0",
-    "@typescript-eslint/parser": "^5.50.0",
+    "@typescript-eslint/eslint-plugin": "^6.7.4",
+    "@typescript-eslint/parser": "^6.7.4",
```
Comment (imjoehaines) [inline package.json:29]: Bumped the `@typescript-eslint` packages for the TypeScript v5 bump

**Review by gingerbenw — APPROVED** [review]

**Pull request merged**